src/MailingListBundle/Block/Service/ContactsManager.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Opifer\MailingListBundle\Block\Service;
+
+use Symfony\Component\DependencyInjection\ContainerInterface as Container;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpKernel\Log\LoggerInterface;
+use Opifer\MailingListBundle\Entity\MailingList;
+use GuzzleHttp\Client as GuzzleClient;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Subscriber\Oauth\Oauth1;
+
+class ContactsManager
+{
+	/** @var string */
+    protected $consumer_key;
+
+    /** @var string */
+    protected $consumer_secret;
+
+    /** @var Service Container */
+    protected $service_container;
+
+    /** @var LoggerInterface */
+    protected $logger;
+
+    public function __construct(EntityManagerInterface $em, Container $container, LoggerInterface $logger, $consumer_key, $consumer_secret)
+    {
+        $this->em = $em;
+        $this->consumer_key = $consumer_key;
+        $this->consumer_secret = $consumer_secret;
+        $this->container = $container;
+        $this->logger = $logger;
+    }
+
+    public function addContactToMailPlus(MailingList $contact)
+    {
+    	if (!empty($contact)) {
+
+    		try {
+
+		    	$stack = HandlerStack::create();
+
+		        $middleware = new Oauth1([
+		            'consumer_key'    => $this->consumer_key,
+		            'consumer_secret' => $this->consumer_secret,
+		            'token'           => '',
+		            'token_secret'    => ''
+		        ]);
+
+		        $stack->push($middleware);
+
+		        $client = new GuzzleClient([
+		            'base_uri' => 'https://restapi.mailplus.nl',
+		            'handler' => $stack,
+		            'auth' => 'oauth',
+		            'headers' => ['Accept' => 'application/json', 'Content-Type' => 'application/json'],
+		        ]);
+
+		        $contact = [
+		            "update" => true,
+		            "purge" => false,
+		            "contact" => [
+		                "externalId"=> $contact->getId(),
+		                //"testGroup"=> true,
+		                "properties"=> [
+		                    "email"=> $contact->getName(),
+		                    "firstName"=> $contact->getDisplayName(),
```

Comment:
There's obviously a misunderstanding here. Here you try to pass the name of the mailinglist as the email of a subscription??

Can you please change this, so it's syncing the correct properties of the correct entity?